### ISSUE TITLE
Add textsring to filename

### ISSUE THREAD

**Issue by unclebrr** [neurobin/rnm#14]
Hello Jahidul,
Thank you for your very useful utility. Now the question:
I have a bunch of fils in dir, like
01-file01.ext
02-file02.ext
(...)
I need to rename them like this:
01-file01 (text).ext
02-file02 (text).ext
(...)
"text" is always the same (see whitespace between filename and (text)).
But when I use -ns '//n/n (text)/g/' I get " (text)" as newname. What I'm doing wrong?
Indexes and so are working good, problem only with textstring. Help!
Thanks in advance.

**Comment by neurobin** [comment]
> -ns '//n/n (text)/g/'

-ns is a name string, in that sense, you have unbalanced `/`

``` sh
-ns '/n/ (text)./e/' 
```

where `/e/` is the extension.

**Comment by unclebrr** [comment]
Thank you! This is SO easy when you know what you are doing :)

**Comment by unclebrr** [comment]
Hello Jahidul Hamid,
Thank you once more for your good program and your help. 
It helps me a lot in my hobbies.
I need another strange thing.
I have a lot of files with a pattern like this:
01_01_longfilemane.anotherpart.extention
01_02_longfilemane.anotherpart.extention
01_03_longfilemane.anotherpart.extention
01_nn_longfilemane.anotherpart.extention
02_01_longfilemane.anotherpart.extention
02_02_longfilemane.anotherpart.extention
02_nn_longfilemane.anotherpart.extention
...
nn_01_longfilemane.anotherpart.extention
nn_nn_longfilemane.anotherpart.extention
and I need to renumber them to
001-01_01.extention
002-01_02.extention
003-01_03.extention
...
nnn-nn_nn.extention

in other words - I want to save first five symbols of the 
file name and renumber them.
I know how to define field length and how to renumber, but 
I don't understand how to leave the part untouched.
Could you help me? Feel stupid not to learn regex...
Thanks in advance.

Mike.
---
Professional hosting for everyone - http://www.host.ru

**Comment by neurobin** [comment]
This should work:

```bash
rnm -ns '/i/-/rn/' -ifl 3 -rs '/^(.{6}).*(\.extention)$/\1\2/' ./*
```

**Comment by unclebrr** [comment]
On Sat, 10 Oct 2020 03:10:15 -0700
  Md Jahidul Hamid <notifications@github.com> wrote:
> This should work:
> rnm -ns '/i/-/rn/' -ifl 3 -rs '/^(.{6}).*(\.extention)$/\1\2/' ./*

No, it simply adds numeration to the files. And I need 
only 4 first symbols of file name.


---
Professional hosting for everyone - http://www.host.ru

**Comment by neurobin** [comment]
I tested it, it does this:

```
01_01_longfilemane.anotherpart.extention --> 001-01_01_.extention
```
Oh, you need `5` instead of `6`, so do:

```bash
rnm -ns '/i/-/rn/' -ifl 3 -rs '/^(.{5}).*(\.extention)$/\1\2/' ./*
```

And use latest version or `rnm`.

**Comment by neurobin** [comment]
By the way, you probably did not change `extention` to the correct extension of the files.

**Comment by neurobin** [comment]
There's another way, where you don't have to edit the command:

```bash
rnm -ns '/i/-/rn/' -ifl 3 -rs '/^(.{5}).*$/\1./e//' ./*
```

**Comment by unclebrr** [comment]
On Sun, 11 Oct 2020 02:41:55 -0700
  Md Jahidul Hamid <notifications@github.com> wrote:
> I tested it, it does this:
> 01_01_longfilemane.anotherpart.extention --> 
>001-01_01_.extention

It's strange because here it does 
001-01_01_longfilemane.anotherpart.extention

> Oh, you need `5` instead of `6`, so do:

Yes, this I can undestand - it is field length :)

> And use latest version or `rnm`.

Hmm. I have this one
rnm -v
Name         : rnm (Bulk rename utility)
Version      : 4.0.9
Author       : Md Jahidul Hamid
Author Email : jahidulhamid@yahoo.com
Bug Report   : https://github.com/neurobin/rnm/issues
---
Professional hosting for everyone - http://www.host.ru

**Comment by unclebrr** [comment]
On Sun, 11 Oct 2020 02:44:46 -0700
  Md Jahidul Hamid <notifications@github.com> wrote:
> By the way, you probably did not change `extention` to 
>the correct extension of the files.

Yes, I didn't :)
Real example:
01_01_Moskva v XVIII stoletii. Vysshee obschestvo. 
Ulichnyy byt naroda. Koronatsiya Ekateriny. 
Sumarokov.wav.new.mp3
I need it to become
01_01.mp3
---
Professional hosting for everyone - http://www.host.ru

**Comment by unclebrr** [comment]
On Sun, 11 Oct 2020 02:51:01 -0700
  Md Jahidul Hamid <notifications@github.com> wrote:
> There's another way, where you don't have to edit the 
>command: 
> rnm -ns '/i/-/rn/' -ifl 3 -rs '/^(.{5}).*$/\1./e//' ./*

Yes! This one works! Thank you!
---
Professional hosting for everyone - http://www.host.ru
